src/index.js

```diff
@@ -63,46 +65,43 @@ module.exports = function (context, options) {
         }
       })
 
-      const versionRegex = options.versionRegex ? new RegExp(options.versionRegex, options.versionRegexOptions) : null;
-
       const addToSearchData = (d) => {
         // read the list of docusaurus versions
-        const docVersions = [
-          ...JSON.parse(
-            fs.readFileSync(
-              path.join(
-                path.resolve(context.siteDir, "docs"),
-                "..",
-                "versions.json"
-              ),
-              "utf-8"
-            )
-          ),
-          "next",
-        ];
+        const versions = JSON.parse(
+          fs.readFileSync(
+            path.join(
+              path.resolve(context.siteDir, "docs"),
+              "..",
+              "versions.json"
+            ),
+            "utf-8"
+          )
+        );
 
         // Determine the version of current document
-        let version = null;
+        // if it hasn't version it's part of the last current public version
+        let version = versions?.length ? "current" : null;
+
+
+        // Retrieve, from plugin options, the Regex expression for getting version info from the path
+        const { versionPathRegex } = options;
 
-        if (versionRegex && docVersions && docVersions.length) {
-          version = docVersions[0];
-          const matchedVersion = d.url.match(versionRegex);
+        if (versionPathRegex && versions && versions.length) {
+          const matchedVersion = d.url.match(versionPathRegex);
           //
-          if (!!matchedVersion && matchedVersion.length) {
+          if (!!matchedVersion && matchedVersion.length)
             version = matchedVersion[1];
-          } else if (d.url.includes("/next/"))
-            version = "next";
         }
-        
-        let lunrAddOptions={
+
+        let lunrAddOptions = {
```

Comment:
This could be a const, and at line 104 it is possible to set the version:
```js
lunrAddOptions.version = version
```

```suggestion
        const lunrAddOptions = {
```